### PR TITLE
Server: Serialize Object controls as JSON over the wire

### PR DESCRIPTION
**Pull request by jonspalmer**
## What I did

Fix the serialization of objects as JSON over the wire that was missed in the refactor of knobs to controls #11658 

Cleaned up some cruft along the way.

### CODE REVIEW

File: app/server/src/client/preview/render.ts [hunk 1]
```diff
@@ -20,17 +20,22 @@ const buildStoryArgs = (args: Args, argTypes: ArgTypes) => {
...
         // use the supplied separator when seriazlizing an array as a string
         const separator = control.separator || ',';
-        storyArgs[key] = storyArgs[key].join(separator);
+        storyArgs[key] = argValue.join(separator);
         break;
       }
+      case 'object':
+        // send objects as JSON strings
+        storyArgs[key] = JSON.stringify(argValue);
+        break;
```
Comment (jonspalmer) [inline app/server/src/client/preview/render.ts:38]: this is the fix

File: examples/server-kitchen-sink/stories/addon-controls.stories.json [hunk 2]
```diff
@@ -34,7 +34,8 @@
         "colour": "deeppink",
         "today": "Jan 20 2017 GMT+0",
         "items":  ["Laptop", "Book", "Whiskey"],
-        "nice": true
+        "nice": true,
+        "other": {"hair": "brown", "eyes": "blue"}
```
Comment (jonspalmer) [inline examples/server-kitchen-sink/stories/addon-controls.stories.json:38]: wasn't testing object controls in the example app

File: examples/server-kitchen-sink/views/addons/controls/all.pug [hunk 3]
```diff
@@ -6,6 +6,7 @@
 - style = `border: 2px dotted ${colour}; padding: 8px 22px; border-radius: 8px`;
 - today = new Date(today);
 - items = items.split(',');
+- other = JSON.parse(other)
```
Comment (jonspalmer) [inline examples/server-kitchen-sink/views/addons/controls/all.pug:9]: proves we have it as json that we can parse

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**